log/handler.go

```diff
@@ -157,9 +158,17 @@ func RotatingFileHandler(path string, limit uint, formatter Format) (Handler, er
 	if counter == nil {
 		counter = new(countingWriter)
 	}
-	h := StreamHandler(counter, formatter)
+	h := FuncHandler(func(r *Record) error {
+		_, err := counter.Write(formatter.Format(r))
+		return err
+	})
 
 	return FuncHandler(func(r *Record) error {
+		// lazy evaluate, this needs no lock.
+		lazyEvaluateRecord(r)
```

Comment:
This is a new addition, right? It wasn't mentioned in the PR description. What was the result before versus after applying this PR?

Reply:
Fixed data race in the original code which uses StreamHandler here.

Reply:
There's a detailed description in the summary part of this PR.

Reply:
Basically, I'm trying to achieve the same effect of `StreamHandler` except move the lock out so that it can protect the `count` and `w` of the `RotatingFileHandler`